### DISCUSSION TITLE
Is there any tutorial on how to use NixOS inside WSL

### DISCUSSION THREAD

**Discussion by xcfw** [nix-community/NixOS-WSL#75] [Q&A] [answered]
Nix noob here, couldn't wrap my head around flake and home-manager. Wiki doesn't mention WSL specific instructions and no info found other than here. My goal was to be able to install a list of apps like `gcloud-cli, kubectl, neovim, miniconda, ruby` etc and to give OS a try.
Somehow installed home-manager (standalone mode, maybe should've gone with NixOS Module approach, please advise) with flakes following this: https://nix-community.github.io/home-manager/index.html#ch-nix-flakes and it seems working, however I'm not sure that it's right way of doing it.

Could you point a documentation that is valid for NixOS under WSL2 for a complete noobs in nix like me?

**Comment by nzbr** [comment, marked as answer]
NixOS-WSL aims to be able to do anything that regular NixOS on hardware can do. To install software, open `/etc/nixos/configuration.nix` in an editor, add the programs to `environment.systemPackages` and run `nixos-rebuild switch`; just like it is described in [the NixOS manual](https://nixos.org/manual/nixos/stable/index.html#sec-declarative-package-mgmt)
